opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -148,10 +139,10 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         if (servletRequest.getAttribute(SERVER_SPAN_CONTEXT) != null) {
             chain.doFilter(servletRequest, servletResponse);
         } else if (isTraced(httpRequest, httpResponse)){
-            SpanContext extractedContext = tracer.extract(Format.Builtin.HTTP_HEADERS,
+            SpanContext extractedContext = GlobalTracer.get().extract(Format.Builtin.HTTP_HEADERS,
                     new HttpServletRequestExtractAdapter(httpRequest));
 
-            final Span span = tracer.buildSpan(httpRequest.getMethod())
+            final Span span = GlobalTracer.get().buildSpan(httpRequest.getMethod())
```

Comment:
I agree with what @yurishkuro said at https://github.com/opentracing-contrib/java-web-servlet-filter/pull/12#issuecomment-286497926

For those who use GlobalTracer, I would expect the following:
- Tracing may not work correctly if the GlobalTracer is changed or reconfigured mid-stream for the application; i.e., it should be set up very early in the process lifetime, and before the process handles its primary transaction workload
- That said, a change to GlobalTracer config should _never_ crash the process / NPE